ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/KeepAlive/Direct.hs

```diff
@@ -1,19 +1,28 @@
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes     #-}
 
 module Ouroboros.Network.Protocol.KeepAlive.Direct where
 
 import           Ouroboros.Network.Protocol.KeepAlive.Client
 import           Ouroboros.Network.Protocol.KeepAlive.Server
 
-direct :: Monad m
+
+direct :: forall a b m. Monad m
```

Comment:
I think we can just use `Applicative`:
```suggestion
direct :: forall a b m. Applicative m
```

Reply:
That doesn't work.

Reply:
Together with the other changes which I suggested down below?